os/kernel/sched/sched_releasetcb.c

```diff
@@ -211,6 +214,18 @@ int sched_releasetcb(FAR struct tcb_s *tcb, uint8_t ttype)
 
 		ret = up_addrenv_detach(tcb->group, tcb);
 #endif
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(CONFIG_ARM_MPU)
+		/* Disable mpu regions when binary is unloaded */
+		if (IS_LOADED_MODULE(tcb->group)) {
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+			for (int i = 0; i < 3 * MPU_NUM_REGIONS; i += 3) {
+				up_mpu_disable_region(&tcb->mpu_regs[i]);
```

Comment:
Please use the up_mpu_check_active() API to check if the mpu region is currently active, before disabling it. Else, sometimes this could disable mpu region of a different app.